src/components/common/Members/Members.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { FC, useCallback, useMemo, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { ColonyRole } from '@colony/colony-js';
+import sortBy from 'lodash/sortBy';
+
+import { SpinnerLoader } from '~shared/Preloaders';
+import {
+  // FormValues as FiltersFormValues,
+  MemberType,
+} from '~common/ColonyMembers/MembersFilter';
+
+import { useColonyContext } from '~hooks';
+import { Domain, useGetMembersForColonyQuery, SortingMethod } from '~gql';
+import {
+  COLONY_TOTAL_BALANCE_DOMAIN_ID,
+  ALLDOMAINS_DOMAIN_SELECTION,
+  ROOT_DOMAIN_ID,
+} from '~constants';
+import { User } from '~types';
+import { notNull } from '~utils/arrays';
+
+import MembersTitle from './MembersTitle';
+// import { filterMembers } from './filterMembers';
+import MembersSection from './MembersSection';
+
+import styles from './Members.css';
+
+const displayName = 'common.Members';
+
+const MSG = defineMessages({
+  loading: {
+    id: `${displayName}.loading`,
+    defaultMessage: "Loading Colony's users...",
+  },
+  failedToFetch: {
+    id: `${displayName}.failedToFetch`,
+    defaultMessage: "Could not fetch the colony's members",
+  },
+  noMembersFound: {
+    id: `${displayName}.noMembersFound`,
+    defaultMessage: 'No members found',
+  },
+});
+
+interface Props {
+  selectedDomain: number | undefined;
+  handleDomainChange: React.Dispatch<React.SetStateAction<number>>;
+  filters: any;
+  // filters: FiltersFormValues;
+}
+
+export type Member = User & {
+  roles: ColonyRole[];
+  directRoles: ColonyRole[];
+  banned: boolean;
+};
+
+const Members = ({ selectedDomain, handleDomainChange, filters }: Props) => {
+  const { colony } = useColonyContext();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [sortingMethod, setSortingMethod] = useState(
+    SortingMethod.ByHighestRep,
+  );
+
+  const { data, loading: loadingMembers } = useGetMembersForColonyQuery({
+    skip: !colony?.colonyAddress,
+    variables: {
+      input: {
+        colonyAddress: colony?.colonyAddress ?? '',
+        domainId: selectedDomain || COLONY_TOTAL_BALANCE_DOMAIN_ID,
+        sortingMethod,
+      },
+    },
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const contributors = useMemo(
+    () => data?.getMembersForColony?.contributors?.filter(notNull) ?? [],
+    [data],
+  );
+  const watchers = useMemo(
+    () => data?.getMembersForColony?.watchers?.filter(notNull) ?? [],
+    [data],
+  );
+
+  // const canAdministerComments =
+  //   hasRegisteredProfile &&
+  //   (hasRoot(currentUserRoles) || canAdminister(currentUserRoles));
+
+  const domainSelectOptions = sortBy(
```

Comment:
This sorting could be wrapped into a utility function `getDomainSelectOptions`, as not easy to see at a glance what's happening here.